client/java/src/main/java/com/linecorp/centraldogma/client/Watcher.java

```diff
@@ -178,4 +182,34 @@ default void watch(Consumer<? super T> listener) {
         requireNonNull(listener, "listener");
         watch((revision, value) -> listener.accept(value));
     }
+
+    /**
+     * @return A new watcher, that reuses the supplied watcher and applies transformation.
+     * Note: the returned watcher is effectively filtering in a sense that,
+     * its listeners are *not* notified when a change has no effect on the transformed value.
+     */
+    default <U> Watcher<U> newChild(Function<T, U> transformer) {
+        return new TransformingWatcher<>(this, transformer);
+    }
+    /**
+     * @return A new child, which transformation is a
```

Comment:
Ditto - We want every Javadoc to have a summary.

Reply:
👍